### PR TITLE
Make growatt play nicer with other modbus components.

### PR DESCRIPTION
**Pull request by onnlucky**
# What does this implement/fix?

This PR makes the growatt_solar component take into account other components using the same modbus component. By only sending a read request if modbus is not waiting for a response. And when receiving data, ignore data that cannot be a response.

## Types of changes

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [X] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
substitutions:
  friendly_name: "Growatt"

# Disable logging to serial, incase the uart is the usb serial, which is by far the easiest for new-ish inverters.
logger:
  baud_rate: 0

uart:
  - id: usb_uart
    baud_rate: 115200
    tx_pin: GPIO1
    rx_pin: GPIO3

modbus:
  id: usb_modbus
  uart_id: usb_uart

modbus_controller:
  - id: usb_modbus_controller
    modbus_id: usb_modbus
    address: 0x1

# Sets up the growatt_solar component
sensor:
  - platform: growatt_solar
    modbus_id: usb_modbus
    address: 0x1
    update_interval: 5s
    protocol_version: RTU2

    energy_production_day:
      name: "${friendly_name} Energy Produced Today"

    total_energy_production:
      name: "${friendly_name} Total Energy Produced"
      accuracy_decimals: 1

# Sets up a modbus_controller component that reads and writes the same device
number:
  - platform: modbus_controller
    modbus_controller_id: usb_modbus_controller
    name: "${friendly_name} Max Active Output Percent"
    id: max_active_output_percent
    register_type: holding
    address: 0x3
    min_value: 0
    max_value: 100
    unit_of_measurement: "%"
    value_type: U_WORD
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/growatt_solar/growatt_solar.h [hunk 1]
```diff
@@ -18,7 +18,12 @@ enum GrowattProtocolVersion {
 };
 
 class GrowattSolar : public PollingComponent, public modbus::ModbusDevice {
+ private:
+  bool waiting_to_update_;
+  uint32_t last_send_;
```
Comment (jesserockz) [inline esphome/components/growatt_solar/growatt_solar.h]: Please move these to `protected`, we generally don't use `private` in ESPHome

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: esphome/components/growatt_solar/growatt_solar.h (hunk 1 above)
Comment (onnlucky) [reply esphome/components/growatt_solar/growatt_solar.h]: Thanks. I've moved them.

**Comment by onnlucky** [comment]
Hi @leeuwte, @jesserockz, or @djwlindenaar, I would love to merge this PR. Can I do anything to help with that?

**Comment by jesserockz** [comment]
@onnlucky Patience is required, you only opened this 5 days ago. There are many PRs that were made before yours that are not merged yet because there are few people reviewing. 

Pinging maintainers will only get your PR ignored until a future date.

Jesse

**Review by jesserockz — APPROVED** [review]

**Pull request merged**